Add support for Elvish transcripts.

xiaq opened the pull request:
This adds support for highlighting transcripts of Elvish sessions.

## Description

Support for Elvish was accepted in https://github.com/github/linguist/pull/5940. Because Elvish is a shell language, it is very frequent to not show just Elvish code, but also its outputs. Instead of just:

```elvish
echo foo
```

One often needs to show:

```elvish-transcript
~> echo foo
foo
```

These _transcripts_ are most often used inside the Markdown of (for example) issue comments, and rarely saved to files, so there aren't a lot of those files on GitHub today.

However, my primary goal is to make it possible to use the `elvish-transcript` language tag on GitHub, and since `elvish` is already usable as a language tag, I'd argue that it should be supported too. The notability test was passed when Elvish itself was accepted.

I did define a file extension for Elvish transcripts, `.elvts`, but I am fine if I should drop that.

## Checklist:

- [x] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - The `.elvts` extension is **not** widely used, see above for explanation.
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - Examples given for the `echo` command (https://elv.sh/ref/builtin.html#echo, [source](https://github.com/elves/elvish/blob/5c835b1c5f5c9576b0d55f001e354f785202a9cf/pkg/eval/builtin_fn_io.d.elv#L160-L167))
    - Sample license(s): BSD 2-clause
  - [x] I have included a syntax highlighting grammar: https://github.com/elves/elvish, already included
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

lildude commented on lib/linguist/languages.yml:
```diff
@@ -1778,6 +1778,16 @@ Elvish:
   tm_scope: source.elvish
   color: "#55BB55"
   language_id: 570996448
+Elvish transcript:
+  type: programming
+  ace_mode: text
+  extensions:
+  - ".elvts"
+  interpreters:
+  - elvish-transcript
+  tm_scope: source.elvish-transcript
+  color: "#99FF99"
+  language_id: 452025714
```
This can be simplified:

```suggestion
Elvish Transcript:
  type: programming
  group: Elvish
  ace_mode: text
  tm_scope: source.elvish-transcript
  color: "#55BB55"
  language_id: 452025714
```

Notable changes:

- we don't need an extension as it'll only ever be used in markdown files
- we group it under Elvish as it is essentially the results of processing that language, like `pycon` for Python
- we don't need an interpreter as this is only used for shebang matching which will never happen
- the colour, whilst never likely to be seen, should probably be the same as Elvish

lildude reviewed (changes requested):
We really don't need an extension or sample in this case and can simplify the entry (see inline comment). 

Widespread usage is still a consideration which is quite easy to determine as you've stated this will only be used in markdown files or issue comments. 

[This search](https://github.com/search?q=language%3Amarkdown+elvish-transcript&type=code) shows it is only being used in 135 markdown files so falls a little short on the files side.

Searching [issue comments or bodies](https://github.com/search?q=%27elvish-transcript%27+in%3Abody%2Ccomments&type=issues) doesn't seem to indicate it's used that much with only 8 results.

So whilst there is usage, it doesn't appear widely adopted on GitHub yet, though I guess the lack of highlighting might have something to do with that in issues.

Let me sleep on this.

xiaq commented:
Thanks, applied your suggested change. Also regenerated vendor/README.md and removed the sample.

I wonder though - how does Linguist know that the `elvish-transcript` language tag in Markdown maps to the `source.elvish-transcript` scope? Is it just based on the name of the scope? (Also FWIW, the reason I added an `interpreter` was based on the misunderstanding that it is also used as a language tag.)

> So whilst there is usage, it doesn't appear widely adopted on GitHub yet, though I guess the lack of highlighting might have something to do with that in issues.

I definitely agree that the lack of usage currently is _because_ there's no syntax highlighting yet, so people don't bother.

Looking at the first page of open issues in https://github.com/elves/elvish/issues, a lot of comments containing transcripts, but not using the `elvish-transcript` tag: https://github.com/elves/elvish/issues/1599#issue-1335425183, https://github.com/elves/elvish/issues/1603#issue-1356716036, https://github.com/elves/elvish/issues/1625#issue-1456592731, https://github.com/elves/elvish/issues/1631#issue-1489888756, https://github.com/elves/elvish/issues/1641#issue-1529522473, https://github.com/elves/elvish/issues/1648#issue-1551873274, https://github.com/elves/elvish/issues/1653#issue-1556086571, https://github.com/elves/elvish/issues/1654#issue-1559100573, https://github.com/elves/elvish/issues/1656#issue-1573626194 (the last code block), https://github.com/elves/elvish/issues/1662#issuecomment-1442799510. That is about 1/3 of all Elvish issues that has at least one transcript and could benefit from `elvish-transcript` support. At least in the context of issues the use of Elvish transcripts is already widespread, just not the use of the `elvish-transcript` tag.

lildude commented:
> I wonder though - how does Linguist know that the `elvish-transcript` language tag in Markdown maps to the `source.elvish-transcript` scope? Is it just based on the name of the scope?

Linguist doesn't directly handle the codeblocks, [markup](https://github.com/github/markup) does, but markup uses the languages and grammars supplied by Linguist. 

I gave a brief run down of how this works just yesterday at https://github.com/github/linguist/issues/6293#issuecomment-1451627502 but in short it knows because users tell it and thanks to the entry in the `languages.yml` file it knows to tell the syntax highlighting engine to use the grammar which has the `source.elvish-transcript` scope to highlight the blocks:

_"The file/codeblock has been identified as 'Elvish Transcript', so Mr Syntax Highlighting Engine, please use the `source.elvish-transcript` grammar to make it pretty"_ 😁

lildude reviewed (approved):
Lets go for it.
